### PR TITLE
Support fastify with nextjs plugin

### PR DESCRIPTION
**Pull request by bbetts-godaddy**
## Summary

Add fastify support in the nextjs plugin

## Changelog

- create fastify hook
- add `nextFastify` lifecycle

## Test Plan
Screen Shot 2022-07-18 at 3 07 10 PM
Screen Shot 2022-07-18 at 3 07 16 PM

### CODE REVIEW

**Comment by jgowdy** [comment]
Logo
**Checkmarx AST – Scan Summary & Details** – [e0bf4a52\-cded\-4d63\-aebd\-f58d6b0d43b0](https://ast.checkmarx.net/projects/fd2d44ed-5f63-497a-b1e1-a5c25aac795a/scans?id=e0bf4a52-cded-4d63-aebd-f58d6b0d43b0&branch=pfx-243)
### New Issues
Severity|Issue|File / Package|Scan Engine
---|---|---|---
HIGH|CVE\-2022\-2216|Npm\-parse\-url\-6\.0\.0|[CxSCA](https://ast.checkmarx.net/sca/#/projects/fd2d44ed-5f63-497a-b1e1-a5c25aac795a/reports/e0bf4a52-cded-4d63-aebd-f58d6b0d43b0/vulnerabilities/all)
HIGH|Cxb3ca64d2\-9cd1|Npm\-mocha\-9\.2\.2|[CxSCA](https://ast.checkmarx.net/sca/#/projects/fd2d44ed-5f63-497a-b1e1-a5c25aac795a/reports/e0bf4a52-cded-4d63-aebd-f58d6b0d43b0/vulnerabilities/all)
MEDIUM|CVE\-2022\-0722|Npm\-parse\-url\-6\.0\.0|[CxSCA](https://ast.checkmarx.net/sca/#/projects/fd2d44ed-5f63-497a-b1e1-a5c25aac795a/reports/e0bf4a52-cded-4d63-aebd-f58d6b0d43b0/vulnerabilities/all)
LOW|Unprotected\_Cookie|/packages/gasket\-plugin\-nextjs/lib/index\.js: [152](https://github.com/godaddy/gasket/blob/pfx-243//packages/gasket-plugin-nextjs/lib/index.js#L152)|[CxSAST](https://ast.checkmarx.net/results/e0bf4a52-cded-4d63-aebd-f58d6b0d43b0/fd2d44ed-5f63-497a-b1e1-a5c25aac795a/sast)

File: packages/gasket-plugin-nextjs/lib/index.js [hunk 1]
```diff
@@ -143,6 +129,37 @@ module.exports = {
...
+      timing: {
+        last: true
+      },
+      handler: async function fastify(gasket, fastifyApp) {
+        const { exec } = gasket;
+        const app = await setupNextApp(gasket);
+
+        fastifyApp.decorate(['buildId', app.name].filter(Boolean).join('/'), {
+          getter() {
+            return app.build;
```
Comment (kawikabader) [inline packages/gasket-plugin-nextjs/lib/index.js:142]: Are we covering this functionality with tests at all?

File: packages/gasket-plugin-nextjs/lib/setup-next-app.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * Small helper function that creates nextjs app from the gasket
+ * configuration.
+ *
+ * @param   {Gasket}  gasket                The gasket API.
+ * @returns {NextServer} The Nextjs App
+ * @private
+ */
+async function setupNextApp(gasket) {
+  const { exec, command } = gasket;
+  const devServer = (command.id || command) === 'local';
```
Comment (kawikabader) [inline packages/gasket-plugin-nextjs/lib/setup-next-app.js:14]: Could we also cover this logic with a quick test?

**Review by kawikabader — COMMENTED** [review]
Logic looks great! Just wanted to see if we could get a few more tests?

**Review by agerard-godaddy — APPROVED** [review]
Looks goods - just expand some of the testing then 👍

**Review by kawikabader — APPROVED** [review]

**Comment by kawikabader** [comment]
Approved! Just may need to resolve a few conflicts before we merge. 👍🏻

**Pull request merged**